### PR TITLE
Pass `Stack` ref to `Completer::fetch`

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Adds an additional `&Stack` parameter to `Completer::fetch` so that the completers don't have to store a `Stack` themselves. I also removed unnecessary  `EngineState`s from the completers, since the same `EngineState` is available in the `working_set.permanent_state` also passed to `Completer::fetch`.

### CODE REVIEW

File: crates/nu-cli/src/completions/base.rs [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+use nu_protocol::{
+    engine::{Stack, StateWorkingSet},
+    levenshtein_distance, Span,
+};
 use reedline::Suggestion;
 
 // Completer trait represents the three stages of the completion
 // fetch, filter and sort
 pub trait Completer {
+    #[allow(clippy::too_many_arguments)]
```
Comment (IanManske) [inline crates/nu-cli/src/completions/base.rs:11]: This is unfortunate, maybe needs revisiting in the future.

**Comment by WindSoilder** [comment]
Not really sure, why store a stack inside `Completer` is a bad idea?

**Comment by IanManske** [comment]
When we do completions, a new completer is created, it is run once with `fetch`, and then it is immediately destroyed. So, rather than taking an owned `Stack`, a `Stack` reference will be sufficient. This avoids cloning the `Stack` for each completion. (The clone should be fast since we have parent `Stack`s now, but no clone is still better than one.) Also, this reduces the lines of code.

**Comment by WindSoilder** [comment]
Thanks!  Then it makes sence to me :)

**Review by WindSoilder — APPROVED** [review]

**Pull request merged**